packages/pwa-kit-extension-sdk/bin/list-overrides.js

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env node
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable @typescript-eslint/no-var-requires */
+const fs = require('fs')
+const path = require('path')
+const chalk = require('chalk')
+
+/**
+ * Recursively lists all files in a directory
+ * @param {string} dir - Directory to scan
+ * @param {string[]} fileList - Accumulator for file list
+ * @returns {string[]} - List of all files in the directory and subdirectories
+ */
+const getAllFiles = (dir, fileList = []) => {
+    if (!fs.existsSync(dir)) {
+        return fileList
+    }
+
+    const files = fs.readdirSync(dir)
+
+    files.forEach((file) => {
+        const filePath = path.join(dir, file)
+        const isDirectory = fs.statSync(filePath).isDirectory()
+
+        isDirectory ? getAllFiles(filePath, fileList) : fileList.push(filePath)
+    })
+
+    return fileList
+}
+
+/**
+ * Reads the overrides stats file and processes the data
+ * @returns {StatsData} Object containing overridable files and stats data
+ */
+const readOverrideStats = () => {
+    const statsPath = path.join(process.cwd(), 'build', 'overrides-stats.json')
+
+    if (!fs.existsSync(statsPath)) {
+        console.error(chalk.red('Error: Stats file not found.'))
+        console.log(
+            chalk.yellow('Please build with RECORD_OVERRIDES=true to generate the stats file:')
+        )
+        console.log(chalk.cyan('RECORD_OVERRIDES=true pwa-kit-dev build'))
+        throw new Error('Stats file not found')
+    }
+
+    try {
+        const statsData = JSON.parse(fs.readFileSync(statsPath, 'utf8'))
+
+        // Extract unique original paths (overridable files)
+        const overridableFiles = statsData.map((entry) => entry.original)
+
+        return {
+            overridableFiles,
+            rawStats: statsData
+        }
+    } catch (error) {
+        throw new Error(`Failed to read override stats: ${error.message}`)
+    }
+}
+
+/**
+ * Checks for unused override files in the overrides directory
+ * @param {StatsData} statsData - Stats data from readOverrideStats
+ */
+const checkUnusedOverrides = ({rawStats}) => {
+    const overridesDir = path.join(process.cwd(), 'app', 'overrides')
+
+    // No overrides directory, nothing to check
```

Comment:
When I ran this on a project without any overrides folder I got this output

```
Listing overridable files...


Overridable files by extension:

No overrides directory found.
```

I wonder if that output could be simplified a little.